onnxruntime/test/optimizer/qdq_transformer_test.cc

```diff
@@ -410,6 +412,44 @@ TEST(QDQTransformerTests, Add_Have_Different_Types) {
   QDQTransformerBinaryOpTests<int8_t, int8_t, int8_t>("Add");
   QDQTransformerBinaryOpTests<int8_t, uint8_t, int8_t>("Add");
   QDQTransformerBinaryOpTests<int8_t, int8_t, uint8_t>("Add");
+  auto test_case = [&](const std::vector<int64_t>& input1_shape, const std::vector<int64_t>& weights_shape) {
+    auto build_test_case = [&](ModelTestBuilder& builder) {
+      auto* input1_arg = builder.MakeInput<float>(input1_shape, -1.f, 1.f);
+      auto* output_arg = builder.MakeOutput();
+
+      // add QDQ activation
+      auto* dq1_output = AddQDQNodePair<int8_t>(builder, input1_arg, .004f, 1);
+
+      // add DQ weights and Add
+      auto* weight = builder.MakeInitializer<uint8_t>(weights_shape, 0, 255);
```

Comment:
we want to check weight type to be S8(int8_t). You can make the activation to be of type uint8_t and the initializer to be of type int8_t.

And could you please extend QDQTransformerBinaryOpTests with an extra template parameter, like IsInput2Constant, instead of adding a lambda function here?